sos/report/plugins/instructlab.py

```diff
@@ -0,0 +1,130 @@
+        PluginOpt('ilab_user', default='', val_type=str,
+                  desc='user that runs instructlab'),
+        PluginOpt('ilab_conf_dir', default='', val_type=str,
+                  desc='instructlab data directory'),
+        PluginOpt('get-cache', default=False,
+                  desc='Capture models and osci cached data')
+    ]
+
+    def setup(self):
+        cont_share_conf_path = '/usr/share/instructlab/config/'
+        cont_opt_path = '/opt/app-root/src'
+        # .cache dir contains the models and oci directories
+        # which can be quite big. We'll gather this only if
+        # specifying it via command line option
+        cache_dir = '/.cache/instructlab'
+        # .config is where the configuration yaml files can
+        # be found. We gather this always.
+        config_dir = '/.config/instructlab'
+        # In the .local directory we can find datasets,
+        # chat logs, taxonomies, and other very useful data
+        # We gather this always.
+        local_share_dir = '/.local/share/instructlab'
+
+        # container paths
+        cont_cache_path = f'{cont_opt_path}{cache_dir}'
+        cont_config_path = f'{cont_opt_path}{config_dir}'
+        cont_local_path = f'{cont_opt_path}{local_share_dir}'
+
+        self.add_forbidden_path([
+            f"{cont_local_path}/taxonomy/.git",
+            f"{cont_local_path}/taxonomy/.github",
+            f"{cont_opt_path}/src/.local/share/instructlab/taxonomy/.git",
+            f"{cont_opt_path}/src/.local/share/instructlab/taxonomy/.github",
+            ])
+
+        subcmds = [
+            'taxonomy diff',
+            'taxonomy diff --taxonomy-base=empty',
+            'system info',
+            'model list',
+            'config show'
+        ]
+
+        data_dirs = [
+            'data',
+            'generated',
+            'taxonomy',
+            'taxonomy_data',
+            'chatlogs',
+            'checkpoints',
+            'datasets',
+            'internal',
+            'phased',
+        ]
+
+        in_container = False
+        container_names = []
+        _containers = self.get_containers()
+
+        for _con in _containers:
+            if _con[1].startswith('instructlab'):
+                in_container = True
+                container_names.append(_con[1])
+
+        if in_container:
+            for cont in container_names:
+                self.add_copy_spec(
+                    [f'{cont_share_conf_path}rhel_ai_config.yaml',
+                     f'{cont_config_path}/config.yaml'],
+                    container=cont)
+                self.add_copy_spec(
+                    [f"{cont_local_path}/{data_dir}"
+                     for data_dir in data_dirs],
+                    container=cont
+                    )
+                self.add_cmd_output(
+                    [f"ilab {sub}" for sub in subcmds],
+                    container=cont
+                )
+                self.add_cmd_output(f"ls -laR {cont_cache_path}",
```

Comment:
can use the new `self.add_dir_listing()` now :)

Reply:
nice! Changing this then